sql/moz-fx-data-shared-prod/udf_js/bootstrap_percentile_ci/udf.sql

```diff
@@ -0,0 +1,212 @@
+/*
+Calculates a confidence interval using an efficient bootstrap sampling technique for a
+given percentile of a histogram.
+
+See:
+- https://mozilla-hub.atlassian.net/browse/DS-2128
+- https://engineering.atspotify.com/2022/03/comparing-quantiles-at-scale-in-online-a-b-testing/
+
+Users must specify an array of percentiles of interest as the first parameter and a
+histogram struct as the second.
+*/
+
+CREATE OR REPLACE FUNCTION udf_js.bootstrap_percentile_ci(
+  percentiles ARRAY<INT64>,
+  histogram STRUCT<values ARRAY<STRUCT<key FLOAT64, value FLOAT64>>>,
+  metric STRING
+)
+RETURNS ARRAY<STRUCT<
+    metric STRING,
+    statistic STRING,
+    point FLOAT64,
+    lower FLOAT64,
+    upper FLOAT64,
+    parameter STRING
+>> DETERMINISTIC
+LANGUAGE js OPTIONS ( library=['gs://moz-data-science-bochocki-bucket/bq_udf_library/qbinom.js'])
```

Comment:
This library needs to be moved to another location, but I'm not sure where to put it.